File: oop/labaratories/labaratory-work-2/Domain/Pizza.cs
namespace labaratory_work_2.Domain;

public enum Dough
{
    Thin,
    Thick
}

public class Pizza : Dish
{
    public Dough Dough { get; set; } = Dough.Thick;

    public Pizza() : base()
    {
    }

    public Pizza(string name) : base(name)
    {
    }

    public Pizza(string name, int weight, double price)
        : base(name, weight, price)
    {
    }

    public void ChangeDough()
    {
        Dough = Dough == Dough.Thin ? Dough.Thick : Dough.Thin;
    }

    public string CutInSlices()
    {
        return $"Cutting the pizza {Name} into slices...";
    }

    public override string DisplayInfo()
    {
        return $"Pizza: {Name}, {Dough} dough, {Weight}g, ${CalcFullPrice()}";
    }

    public override double CalcFullPrice()
    {
        return Math.Round(
            Dough == Dough.Thin ? Price * 1.3 : Price * 1.5,
            2,
            MidpointRounding.AwayFromZero
        );
    }
}

File: oop/lab-3/src/main/java/com/example/demo/domain/Pizza.java
package com.example.demo.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import java.math.BigDecimal;
import java.math.RoundingMode;

@Entity
@Table(name = "pizzas")
@PrimaryKeyJoinColumn(name = "dish_id")
@OnDelete(action = OnDeleteAction.CASCADE)
@Getter @Setter
public class Pizza extends Dish {
    public enum Dough { THIN, THICK }

    @Enumerated(EnumType.STRING)
    private Dough dough = Dough.THICK;

    public Pizza() { super(); }
    public Pizza(String name) { super(name); }
    public Pizza(String name, int weight, double price) { super(name, weight, price); }

    public void changeDough() {
        this.dough = (this.dough == Dough.THIN) ? Dough.THICK : Dough.THIN;
    }

    public String cutInSlices() {
        return "Cutting the pizza " + getName() + " into slices...";
    }

    @Override
    public String displayInfo() {
        return String.format("Pizza: %s, %s dough, %dg, $%s",
                getName(), getDough(), getWeight(), calcFullPrice());
    }

    @Override
    public double calcFullPrice() {
        double factor = (this.dough == Dough.THIN) ? 1.3 : 1.5;
        double result = getPrice() * factor;
        return BigDecimal.valueOf(result)
                .setScale(2, RoundingMode.HALF_UP)
                .doubleValue();
    }
}